lib/src/tasks/coverage/api.dart

```diff
@@ -537,8 +550,32 @@ class CoverageTask extends Task {
           break;
         }
       }
+      if (server != null && server.observatoryPort.length > 0) {
+        List<int> result = await server.checkPorts();
+        return result;
+      }
+      return [port];
+    }
+  }
 
-      return port;
+  void _testFileValidation(String path, List<File> fileCollection) {
```

Comment:
This is a bit of an anti-pattern - I would expect this method to return a list of files instead of populating a list that gets passed in.